src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/PackageSourceMoniker.cs

```diff
@@ -111,7 +111,7 @@ public static ValueTask<IReadOnlyCollection<PackageSourceMoniker>> PopulateListA
                 .ToList();
 
             var packageSourceMonikers = new List<PackageSourceMoniker>();
-            if (packageSources.Count > 1) // If more than 1, add 'All'
+            if (enabledSources.Count > 1) // If more than 1, add 'All'
```

Comment:
Should this condition apply to https://github.com/NuGet/NuGet.Client/pull/3886/files#diff-01e69bcda3a77f04e4b28e6fef0b520fd4d319c3fc2c83a1930d757d47768823R119 ?
Seems the same exception (assuming ArgumentNull) would occur on the very next line.

Reply:
packageSources is not used after line 109, the problem also wasn't that packageSources was null it was that all the sources were disabled so enabledSources was an empty list

packageSources should theoretically never be null since we will always have VS offline cache that the user cannot remove through UI

Reply:
Sorry if GH isn't making this clear, but I commented on the line where you switched to `enabledSources`. I also linked a line of code below where `enabledSources` is used in a LINQ expression. It would make sense that we not invoke LINQ on a collection we know is empty already.

Reply:
I was referring to `enabledSources`, but actually looking at this now, I have no idea what I was seeing/thinking yesterday.